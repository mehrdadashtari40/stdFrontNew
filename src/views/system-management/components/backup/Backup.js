import React from 'react'
import {config} from '../../../../config/config';
import getAuthonticatedJSON from '../../../../common/utils/functions/getAuthenticatedJSON';
import {withTranslation} from 'react-i18next';
import CardContent from "@mui/material/CardContent";
import Card from "@mui/material/Card";
import postAuthonticatedJSON from "../../../../common/utils/functions/postAuthenticatedJSON";

class Backup extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            spinner: false,
            download_link:''
        }
    }

    backup = () => {
        this.setState({spinner: true});
        let self = this;
        let temp_items = config.apiServer.split('/');
        let workspace = temp_items[temp_items.length - 2];
        let URL = config.apiServer + 'arian/ut/backup-with-link/' + workspace;

        postAuthonticatedJSON(URL)
            .then((res) => {
                self.setState({spinner: false,download_link:res});
            }).catch(() => {
            self.setState({spinner: false});
        });
    }

    downloadFile = (filename, text) => {
        var element = document.getElementById('export');
        // element.setAttribute('href', 'data:text/plain;charset=utf-8,' + encodeURIComponent(text));
        element.setAttribute('href', 'data:application/x-gzip;charset=utf-8,' + encodeURIComponent(text));
        element.setAttribute('download', filename);

        element.style.display = 'none';
        // document.body.appendChild(element);
        element.click();
    }

    render() {
        let spinner = null;
        if (this.state.spinner)
            spinner = <i className="fa fa-spin fa-spinner" style={{marginLeft: '5px'}}></i>

        return (<>
            <Card id="content">
                <CardContent>
                    <h5 className={'table-header-title'}>
                        <i className={"fal fa-download table-header-icon"}></i>
                        تنظیمات - نسخه پشتیبان
                    </h5>
                    <div className="row">
                        <div className="col-lg-12">
                            <div className="center" style={{textAlign: 'center'}}>
                                <br/>
                                {this.state.download_link === ""?
                                    <button className="btn btn-primary" onClick={this.backup}>
                                    {spinner}
                                    ایجاد لینک دانلود
                                </button>:<a className="btn btn-success" href={config.apiServer + 'arian/ut/backup-with-link/' + this.state.download_link}>دانلود فایل</a> }

                                <a id="export" className="hidden">export file</a>
                                <br/>
                                <br/>
                            </div>
                        </div>
                    </div>
                </CardContent>
            </Card>
        </>)
    }
}

export default withTranslation()(Backup);