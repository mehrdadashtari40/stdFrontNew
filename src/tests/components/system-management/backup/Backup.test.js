import React from 'react';
import { shallow } from 'enzyme';
import Backup from '../../../../views/system-management/components/backup/Backup';
jest.mock('react-i18next', () => ({
	withTranslation: () => Component => {
		Component.defaultProps = { ...Component.defaultProps, t: () => '' };
		return Component;
	},
}));

test('Make Snapshot file , Backup Component', () => {
	const wrapper = shallow(<Backup />);
	expect(wrapper).toMatchSnapshot();
});

it('backup HaveBeenCalled ?', () => {
	const wrapper = shallow(<Backup />);

	const instance = wrapper.instance();
	const spy = jest.spyOn(instance, 'backup');

	instance.forceUpdate();

	const Button = wrapper.find('button');
	Button.simulate('click');
	expect(spy).toHaveBeenCalled();
});

test('backup Function  ', () => {
	const wrapper = shallow(<Backup />);
	const instance = wrapper.instance();
	expect(instance.backup).toBeDefined();
});

test('downloadFile Function  ', () => {
	const wrapper = shallow(<Backup />);
	const instance = wrapper.instance();
	expect(instance.downloadFile).toBeDefined();
});
