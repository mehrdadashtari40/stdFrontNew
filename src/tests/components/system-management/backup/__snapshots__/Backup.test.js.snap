// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Make Snapshot file , Backup Component 1`] = `
<WidgetGrid>
  <div
    id="content"
  >
    <div
      className="inbox-nav-bar no-content-padding"
    >
      <div
        className="big-breadcrumbs"
      >
        <h1
          className="page-title txt-color-blueDark"
        >
          <i
            className="fa fa-fw fa-inbox"
          />
          تنظیمات
          <span>
            <span
              className="page-title-separator"
            >
              <i
                className="fa fa-angle-right"
              />
                
            </span>
            نسخه پشتیبان
          </span>
        </h1>
      </div>
      <div
        className="row"
      >
        <article
          className="col-sm-12 col-md-12 col-lg-12"
        >
          <div
            className="inbox-checkbox-triggered"
          >
            <JarvisWidget
              collapsed={false}
              color={false}
              colorbutton={false}
              custombutton={false}
              deletebutton={false}
              editbutton={false}
              fullscreenbutton={true}
              hidden={false}
              load={false}
              refresh={false}
              sortable={true}
              togglebutton={false}
            >
              <header>
                <nav
                  className="inbox_header_item_bx"
                >
                  <p />
                </nav>
              </header>
              <div
                className="inbox-body1 no-content-padding1 no-padding"
                id="inbox-content"
              >
                <fieldset>
                  <section>
                    <div
                      className="row"
                    >
                      <div
                        className="col-lg-12"
                      >
                        <div
                          className="center"
                          style={
                            Object {
                              "textAlign": "center",
                            }
                          }
                        >
                          <br />
                          <button
                            className="btn btn-primary"
                            onClick={[Function]}
                          >
                            دانلود فایل
                          </button>
                          <a
                            className="hidden"
                            id="export"
                          >
                            export file
                          </a>
                          <br />
                          <br />
                        </div>
                      </div>
                    </div>
                  </section>
                </fieldset>
              </div>
              <a
                className="btn btn-primary pull-right hidden-desktop visible-tablet"
                id="compose-mail-mini"
              >
                <strong>
                  <i
                    className="fa fa-file fa-lg"
                  />
                </strong>
              </a>
            </JarvisWidget>
          </div>
        </article>
      </div>
    </div>
  </div>
</WidgetGrid>
`;
